__device-tests__/gutenberg-editor-writing-flow.test.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Internal dependencies
+ */
+const { blockNames } = editorPage;
+
+describe( 'Gutenberg Editor - Writing Flow', () => {
+	describe( 'TC007 - Test format detection under the cursor', () => {
+		it( 'checks that the proper format buttons are selected when the cursor is under', async () => {
+			const paragraphHTML = `<!-- wp:paragraph -->
+            <p><span accessibility-id="bold"><strong>bold</strong></span>
+            <span accessibility-id="bold-italic"><strong><i>bold-italic</i></strong>.</span>
+            <span accessibility-id-"strikethrough"><s>strikethrough</s></span></p>
+            <!-- /wp:paragraph -->`;
+
+			// On a rich-text based component, add bold, italic, strikethrough and link formatted text
+			await editorPage.addNewBlock( blockNames.paragraph );
+			const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.paragraph
+			);
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphHTML
+			);
+
+			// Select bold text
+			// TODO: verify if this is the best method to select text formatting
+			await editorPage.elementsByAccessibilityId( 'bold' );
```

Comment:
If we want to verify the resulting markup is accurate, we might should rely upon [`getHtmlContent`](https://github.com/WordPress/gutenberg/blob/dfdfaa3135b32af9018a41b832c2538c49adf747/packages/react-native-editor/__device-tests__/pages/editor-page.js#L261-L269) and assert with a [Jest snapshot](https://jestjs.io/docs/snapshot-testing). 

That said, a question worth asking is whether this test case should focus on markup or merely the status of formatting buttons. We may not need to assert that the markup is correct within this test. WDYT?

Reply:
> If we want to verify the resulting markup is accurate, we might should rely upon [getHtmlContent](https://github.com/WordPress/gutenberg/blob/dfdfaa3135b32af9018a41b832c2538c49adf747/packages/react-native-editor/__device-tests__/pages/editor-page.js#L261-L269) and assert with a [Jest snapshot](https://jestjs.io/docs/snapshot-testing).

In the code comment when I referenced "select text formatting", I was referring to the actual act of pressing the text formatting buttons themselves. In that context, the "select" verb referred to getting some sort of testID or accessibility label for the bold, italic, strikethrough buttons (etc) in the bottom button bar. However, after reading your [comment](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5610#pullrequestreview-1365244824) about reviewing the multitude of available test helpers to gain more insight, I became aware of the [`toggleFormatting`](https://github.com/WordPress/gutenberg/blob/dfdfaa3135b32af9018a41b832c2538c49adf747/packages/react-native-editor/__device-tests__/pages/editor-page.js#L585-L593) helper, which I believe will solve what my initial TODO comment was trying to verify. I've updated the test to type plain text and toggle the formatting buttons:

```
await editorPage.typeTextToTextBlock(
	paragraphBlockElement,
	'plaintext string'
);

// Toggle various formatting options, then proceed with typing more text
await editorPage.toggleFormatting( 'Bold' );
```

However, you raise a good points about whether or not we should verify the markup is accurate with an assertion:

> That said, a question worth asking is whether this test case should focus on markup or merely the status of formatting buttons. We may not need to assert that the markup is correct within this test. WDYT?

I definitely think checking the HTML and asserting the value would be helpful here. When I perform this test manually, I always switch to HTML mode to verify with the notion that the markup we're "sending" to the web is accurate and valid. I think replicating this step in the automated test would be valuable, so I'll update that as well, and thanks for the call-out. 👍

Reply:
> My only objection to asserting the markup in this test is that we could likely assert the markup output in an integration test, or we may already do so now.

That makes sense. I'll verify if this is already covered in an integration test, and assert here if not, or otherwise utilize the snapshots instead.